common/utils.go

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-mysql-org/go-mysql/canal"
+	"github.com/jmoiron/sqlx"
+)
+
+func FormatValue(val any) any {
+	switch val := val.(type) {
+	case time.Time:
+		return val.UTC().Format(time.RFC3339)
+	case *time.Time:
+		return val.UTC().Format(time.RFC3339)
+	case int8:
+		return int(val)
+	case int16:
+		return int(val)
+	case int32:
+		return int(val)
+	case int64:
+		return int(val)
```

Comment:
This will cause loss of precision.

Reply:
Thanks, I remember adding this because how mysql canal did format values. I'll remove this in this pr and try to merge this change into next prs.

Reply:
Also @lyuboxa, I remember having discrepancies with how the mysql canal and the driver parsed ints. The driver did offer me an int32 for the user id, and the canal did offer me an int. That's why I added the int conversion.

Should I add more logic into the tests (to compare different types of ints) rather than what I was doing here?

Reply:
Oh nvm, found the better solution. I'll just use `JSONBytesEqual` utility I made to assert contents, which will ignore the size of the int when comparing data (as long as they are the same number ofc).

Reply:
Do you mean `canal` will pass int64 as ints? because that is a problem.

JSON has the same issue, if a JSON value has an int64 JSON will shove it in a float64 which will lose precision. 
`json.Number` should be used to deal with that. Which is really bytes/string representation ;-)

Reply:
@lyuboxa I was wrong, canal actually gives me an int64 for the user.id column, sorry for misguiding you. The problem I was having was that the `testutils User` struct had an ID field that had to be both int32 for the snapshot tests and int64 for the cdc tests, which obviously wasn't possible.

> JSON has the same issue, if a JSON value has an int64 JSON will shove it in a float64 which will lose precision.
json.Number should be used to deal with that. Which is really bytes/string representation ;-)

The only place in the code where I see that that can happen is when I marshal the snapshot key. I'll create an issue to see if I can handle that edge case.

Reply:
Actually this is only an issue when deser into `any` without specific type, it will default to float64.
If you are deser into an int64 it will do the right thing. 

See https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/encoding/json/decode.go;l=989-995